distributed/shuffle/_scheduler_plugin.py

```diff
@@ -50,8 +51,14 @@ def to_msg(self) -> dict[str, Any]:
     def __str__(self) -> str:
         return f"{self.__class__.__name__}<{self.id}[{self.run_id}]>"
 
+    def __eq__(self, other: Any) -> bool:
+        return type(other) == type(self) and other.run_id == self.run_id
```

Comment:
This appears to be not covered by tests. Why is this necessary then?

Reply:
It had some use in an earlier iteration. Removed.

Reply:
> I suppose this is because the run_id is a unique token that defines the shuffle state object.

That's why I initially had a new `__eq__`. Then `__hash__` had to match it. Now I'm only using `__hash__`, so I _think_ there's no need for a custom `__eq__` that could potentially get outdated.